python/ray/dag/compiled_dag_node.py

```diff
@@ -958,23 +965,19 @@ def _preprocess(self) -> None:
         # custom NCCL groups for a set of actors, only one is cached.
         actors_to_nccl_group_id: Dict[FrozenSet["ray.actor.ActorHandle"], str] = {}
 
-        # If a custom NCCL group is specified for P2P actors, initialize and cache
-        # the NCCL group ID.
-        if nccl_actors_p2p and self._custom_nccl_group_p2p:
-            if not set(nccl_actors_p2p).issubset(
-                set(self._custom_nccl_group_p2p.get_actor_handles())
-            ):
-                raise ValueError(
-                    "Expected P2P actor handles to be a subset of the custom NCCL group"
-                )
-            self._nccl_group_id_p2p = _init_nccl_group(
-                nccl_actors_p2p, self._custom_nccl_group_p2p
+        for custom_nccl_group, dag_nodes in custom_nccl_group_to_dag_nodes.items():
```

Comment:
Need comments for this, similar to other blocks.

Reply:
Nit: `dag_nodes` can be reduced to `nodes`.

Reply:
Added comment.

Reply:
Changed `dag_node` to `node`.